### DISCUSSION TITLE
TransformBounds Method equivalent

### DISCUSSION THREAD

**Discussion by workgroupengineering** [AvaloniaUI/Avalonia#11382] [Q&A] [answered]
What is equivalent of the following code?

```csharp
   var associatedElementRect = AssociatedObject!.TransformToVisual(_hostScrollViewer!)
        .TransformBounds(new Rect(0, 0, AssociatedObject.ActualWidth, AssociatedObject.ActualHeight));
```

Avalonia:  *11.0.0-Preview8*

**Comment by wieslawsoltes** [comment]
https://github.com/AvaloniaUI/Avalonia/blob/3aa881b0075ea6b7866084de234754fe4985342d/src/Avalonia.Base/VisualTree/VisualExtensions.cs#L207

**Reply by workgroupengineering** [reply]
Thanks, I've already seen it. I'm trying to port [ViewportBehavior](https://github.com/CommunityToolkit/WindowsCommunityToolkit/blob/main/Microsoft.Toolkit.Uwp.UI.Behaviors/Viewport/ViewportBehavior.cs). The problem with this api is that it returns as relative to the first null visual parent and not the specified parent. 
Do you know if it exists? 
If it doesn't exist, can I do a PR?

**Reply by workgroupengineering** [reply, marked as answer]
I have resolevd with this:

```csharp
    public static Rect? TransformBoundsTo(this Visual visual, Visual relativeTo)
    {
        if (visual.TransformToVisual(relativeTo) is Matrix trasformation)
        {
            var p0 = trasformation.Transform(default);
            var p1 = trasformation.Transform(new(visual.Bounds.Width, visual.Bounds.Height));
            return new(p0, p1);
        }
        return default;
    }
```
